Police-Data-Accessibility-Project/data-sources-app-v2
# Changes for 299 - Github Issue Posting

### Fixes

* https://github.com/Police-Data-Accessibility-Project/data-sources-app/issues/464
* Part of https://github.com/Police-Data-Accessibility-Project/data-sources-app/issues/299

### Description

* Add Endpoint for adding Data Request to Github Issues, and associated logic and tests
* Add `PyGithub` as a dependency 

#### LIBTYFI:
* Create `third_party_interaction_logic` directory
* Move `callback_` logic to relevant directories
* Update `PsycopgResources.handle_exceptions` to simplify logic
* Update `test_data_creator_flask` to propagate exceptions on error
* Create `manual_tests` directory

### Testing

* Run tests in `tests` directory to confirm functionality 
* Inspect API documentation to confirm expected appearance and behavior

### Performance

* Time required to post to Github issue is minor
* All other performance impacts are in line with performance of other endpoints that interface with database.

### Docs

* API documentation updated

### Breaking Changes

* No breaking changes in happy path of logic
* Change to `handle_exceptions` logic may change how unexpected errors appear in the response. However, this is marginal as no logic is designed to account for this.
